From c75f073514d08be14858e4f4e72711f266d2b523 Mon Sep 17 00:00:00 2001
From: wilson_chen <wilson_chen@phoenix.com>
Date: Wed, 15 May 2024 17:56:25 +0800
Subject: [PATCH] 1. With the Command Prompt of [Developer Command Prompt for
 VS2015] ("C:\Program Files (x86)\Microsoft Visual Studio
 14.0\Common7\Tools\VsDevCmd.bat"), launch: >edksetup.bat

2. Changed 'TOOL_CHAIN_TAG' in [$(WORKSPACE)\Conf\target.txt] as:
TOOL_CHAIN_TAG = VS2015

3. Start the build by launch:
>Build

ERROR: Would get following build error message:
'c:\Program' is not recognized as an internal or external command,
operable program or batch file.
NMAKE : fatal error U1077: '"c:\Program Files\Windows Kits\8.1\bin\x86\\rc.exe' : return code '0x1'
Stop.

Fix the build error,
Tested :
  TOOL_CHAIN_TAG = VS2015
  TOOL_CHAIN_TAG = VS2015x86

Signed-off-by: wilson_chen <wilson_chen@phoenix.com>
---
 BaseTools/Conf/tools_def.template |  4 ++--
 BaseTools/set_vsprefix_envs.bat   | 12 ++++++++++--
 2 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/BaseTools/Conf/tools_def.template b/BaseTools/Conf/tools_def.template
index c34ecfd557..71d1b96fed 100755
--- a/BaseTools/Conf/tools_def.template
+++ b/BaseTools/Conf/tools_def.template
@@ -59,8 +59,8 @@ DEFINE WINSDK_BIN           = ENV(WINSDK_PREFIX)
 DEFINE WINSDKx86_BIN        = ENV(WINSDKx86_PREFIX)
 
 # Microsoft Visual Studio 2015 Professional Edition
-DEFINE WINSDK81_BIN         = ENV(WINSDK81_PREFIX)x86\
-DEFINE WINSDK81x86_BIN      = ENV(WINSDK81x86_PREFIX)x64
+DEFINE WINSDK81_BIN         = ENV(WINSDK81_PREFIX)
+DEFINE WINSDK81x86_BIN      = ENV(WINSDK81x86_PREFIX)
 
 # Microsoft Visual Studio 2017/2019 Professional Edition
 DEFINE WINSDK10_BIN         = ENV(WINSDK10_PREFIX)DEF(VS_HOST)
diff --git a/BaseTools/set_vsprefix_envs.bat b/BaseTools/set_vsprefix_envs.bat
index 0b9a0c75b9..98e177f0f6 100644
--- a/BaseTools/set_vsprefix_envs.bat
+++ b/BaseTools/set_vsprefix_envs.bat
@@ -34,10 +34,18 @@ if defined VS140COMNTOOLS (
     set "VS2015_PREFIX=%VS140COMNTOOLS:~0,-14%"
   )
   if not defined WINSDK81_PREFIX (
-    set "WINSDK81_PREFIX=c:\Program Files\Windows Kits\8.1\bin\"
+    if exist "%ProgramFiles(x86)%\Windows Kits\8.1\bin" (
+      set "WINSDK81_PREFIX=%ProgramFiles(x86)%\Windows Kits\8.1\bin\"
+    ) else if exist "%ProgramFiles%\Windows Kits\8.1\bin" (
+      set "WINSDK81_PREFIX=%ProgramFiles%\Windows Kits\Windows Kits\8.1\bin\"
+    )
   )
   if not defined WINSDK81x86_PREFIX (
-    set "WINSDK81x86_PREFIX=c:\Program Files (x86)\Windows Kits\8.1\bin\"
+    if exist "%ProgramFiles(x86)%\Windows Kits\8.1\bin" (
+      set "WINSDK81x86_PREFIX=%ProgramFiles(x86)%\Windows Kits\8.1\bin\"
+    ) else if exist "%ProgramFiles%\Windows Kits\8.1\bin" (
+      set "WINSDK81x86_PREFIX=%ProgramFiles%\Windows Kits\Windows Kits\8.1\bin\"
+    )
   )
 ) else (
   if /I "%1"=="VS2015" goto ToolNotInstall
-- 
2.35.1.windows.2

